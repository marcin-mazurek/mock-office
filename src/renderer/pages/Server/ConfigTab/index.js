import React from 'react';
import { Chip } from 'react-mdl';
import FilePicker from '../../../expectations/addFromFile/FilePicker';
import HttpLoadedExpectationList from '../HttpLoadedExpectationList';
import HttpUnloadedExpectationList from '../HttpUnloadedExpectationList';
import WsLoadedExpectationList from '../WsLoadedExpectationList';
import WsUnloadedExpectationList from '../WsUnloadedExpectationList';
import StartButton from '../../../servers/startServer/StartButton';
import StopButton from '../../../servers/stopServer/StopButton';

const ConfigTab = ({ running, serverDetails }) => {
  const { port, type } = serverDetails;

  return (
    <div>
      <div className="server-config-tab">
        <div className="server-config-tab-details__row">
          <div className="server-config-tab-details__cell">
            <Chip>
              Port: {port}
            </Chip>
          </div>
          <div className="server-config-tab-details__cell">
            <Chip>
              Type: {type}
            </Chip>
          </div>
        </div>
        <div className="server-config-tab-details__row">
          <div className="server-config-tab-details__cell">
            { running ? <StopButton /> : <StartButton /> }
          </div>
          <div className="server-config-tab-details__cell">
            <FilePicker />
          </div>
        </div>
      </div>
      <div className="server-config-tab__expectations server-config-tab-expectations">
        <div className="server-config-tab-expectations-type server-config-tab-expectations__type">
          <span>Not loaded:</span>
          <div className="server-config-tab-expectations-type__list">
            { type === 'http' ? <HttpUnloadedExpectationList /> : null }
            { type === 'ws' ? <WsUnloadedExpectationList /> : null }
          </div>
        </div>
        <div className="server-config-tab-expectations-type server-config-tab-expectations__type">
          <span>Loaded:</span>
          <div className="server-config-tab-expectations-type__list">
            { type === 'http' ? <HttpLoadedExpectationList /> : null }
            { type === 'ws' ? <WsLoadedExpectationList /> : null }
          </div>
        </div>
      </div>
    </div>
  );
};

ConfigTab.propTypes = {
  running: React.PropTypes.bool.isRequired,
  serverDetails: React.PropTypes.shape({
    name: React.PropTypes.string.isRequired,
    type: React.PropTypes.string.isRequired,
    port: React.PropTypes.number.isRequired
  })
};

export default ConfigTab;
