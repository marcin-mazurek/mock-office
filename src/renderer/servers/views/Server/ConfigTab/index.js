import React from 'react';
import { Chip } from 'react-mdl';
import FilePicker from '../../../../queues/addFromFile/FilePicker';
import Queues from '../Queues';

const ConfigTab = ({ serverDetails }) => {
  const { port, type } = serverDetails;

  return (
    <div className="server-config-tab">
      <div className="server-config-tab-details">
        <div className="server-config-tab-details__row">
          <div className="server-config-tab-details__cell">
            <Chip>
              Port: {port}
            </Chip>
          </div>
          <div className="server-config-tab-details__cell">
            <Chip>
              Type: {type}
            </Chip>
          </div>
        </div>
        <div className="server-config-tab-details__row">
          <div className="server-config-tab-details__cell">
            <FilePicker />
          </div>
        </div>
      </div>
      <div className="server-config-tab__expectations server-config-tab-queues">
        <Queues />
      </div>
    </div>
  );
};

ConfigTab.propTypes = {
  serverDetails: React.PropTypes.shape({
    name: React.PropTypes.string.isRequired,
    type: React.PropTypes.string.isRequired,
    port: React.PropTypes.number.isRequired
  })
};

export default ConfigTab;
