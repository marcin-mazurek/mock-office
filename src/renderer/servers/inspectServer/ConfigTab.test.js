import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConfigTab from './ConfigTab';

test('ConfigTab should render', () => {
  const serverDetails = {
    name: 'some name',
    type: 'some type',
    queue: 'queue id',
    id: 'some id',
    port: 3000
  };
  const wrapper = shallow(<ConfigTab serverDetails={serverDetails} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
