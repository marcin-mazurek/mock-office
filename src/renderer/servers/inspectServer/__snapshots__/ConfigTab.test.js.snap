exports[`test ConfigTab should render 1`] = `
<div
  className="server-config-tab">
  <div
    className="server-config-tab-details">
    <div
      className="server-config-tab-details__row">
      <div
        className="server-config-tab-details__cell">
        <Chip>
          Port: 
          3000
        </Chip>
      </div>
      <div
        className="server-config-tab-details__cell">
        <Chip>
          Type: 
          some type
        </Chip>
      </div>
    </div>
    <div
      className="server-config-tab-details__row">
      <div
        className="server-config-tab-details__cell">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/add-task">
          Add task
        </Link>
      </div>
    </div>
    <div
      className="server-config-tab-details__row">
      <div
        className="server-config-tab-details__cell">
        Add from file: 
        <Connect(FilePicker) />
      </div>
    </div>
  </div>
  <div
    className="server-config-tab__tasks server-config-tab-queues">
    <Component
      id="queue id"
      serverId="some id" />
  </div>
</div>
`;
